pkg/signature/kms/cliplugin/signer_program_test.go

```diff
@@ -0,0 +1,146 @@
+//
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package cliplugin implements the plugin functionality.
+
+//go:build signer_program
+// +build signer_program
+
+package cliplugin
+
+import (
+	"context"
+	"crypto"
+	"flag"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/sigstore/sigstore/pkg/signature/kms/cliplugin/common"
+)
+
+// This file tests the PluginClient against a pre-built plugin programs.
+// It is skipped during normal `go test ./...` invocations. It can be invoked like
+// `go test -tags=signer_program -count=1 ./... -key-resource-id [my-kms]://[my-key-ref]`
+// See ./README.md for plugin program usage.
+
+var (
+	inputKeyResourceID = flag.String("key-resource-id", "", "key resource ID for the KMS, defaults to 'testkms://testkey'")
+	testHashFunc       = crypto.SHA512
+)
+
+func getPluginClient(t *testing.T) *PluginClient {
+	t.Helper()
+	executable, keyResourceID, err := getPluginExecutableAndKeyResourceID(*inputKeyResourceID)
+	if err != nil {
+		t.Fatalf("build flag -key-resource-id is required for this test: %v", err)
+	}
+	initOptions := &common.InitOptions{
+		ProtocolVersion: common.ProtocolVersion,
+		KeyResourceID:   keyResourceID,
+		HashFunc:        testHashFunc,
+		// TODO: include extracted values from opts
+	}
+	pluginClient := newPluginClient(executable, initOptions)
+	return pluginClient
+}
+
+func Test_DefaultAlgorithm(t *testing.T) {
```

Comment:
```suggestion
func TestDefaultAlgorithm(t *testing.T) {
```

Reply:
No need for the underscore in test names

Reply:
fixed.